### PR TITLE
Adding another flavour of VIL to RRFS for FAA users.

### LINKED ISSUE #799: Add RADARVIL for RRFS
FAA colleagues have requested both radar-based on hydrometeor-based vertically integrated liquid (VIL) from RRFS.  At the moment, we only output one of the two types.

### LINKED ISSUE #804: fv3atm using current fae617b hash of upp
I'm trying to update branches in my fv3atm fork and I keep getting a conflict : 
``modified:   parm/postxconfig-NT-gefs-aerosol.txt``

The UPP hash was updated yesterday to fae617b for UPP.

If I try to check out the UPP (this repo) locally, I get:

warning: the following paths have collided (e.g. case-sensitive paths
on a case-insensitive filesystem) and only one from the same
colliding group is in the working tree:

  'parm/postxconfig-NT-GEFS-aerosol.txt'
  'parm/postxconfig-NT-gefs-aerosol.txt'

Then I see

```
~/GitHub/UPP/parm % git status
On branch develop
Your branch is up to date with 'origin/develop'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   postxconfig-NT-GEFS-aerosol.txt

no changes added to commit (use "git add" and/or "git commit -a")
```

I get the lower case gefs when I checkout the fv3atm repo. But this file shows differences with the lower case gefs file from the UPP repo.

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
FAA users have requested both hydrometeor-based and radar-based VIL from RRFS.  We were previously only providing RADARVIL; this PR enables output of hydrometeor-based VIL.  

The change was tested for RRFS_CONUS_3km on Jet.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Can you sync your branch with latest UPP develop branch?

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA From my testing, I saw the new field hydrometeor-based VIL generated in RRFS products as:
```
[Wen.Meng@hfe06 fv3r_2023062800]$ wgrib2 PRSLEV10.tm00 -match VIL -s -stats
41:39127273:d=2023062800:VIL:entire atmosphere:10 hour fcst::ndata=1905141:undef=0:mean=0.0998323:min=0:max=92.7872:cos_wt_mean=0.097041
```
You may access the RRFS test case at /scratch1/NCEPDEV/stmp2/Wen.Meng/fv3r_2023062800 on Hera. Please let me know if you see inconsistency with your testing.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs were completed on WCOSS2. There will be baseline recreation for fv3r and 3drtma in this PR.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA You may start the UPP RTs on Hera and Orion. Please refer to my testing on Hera  /scratch1/NCEPDEV/stmp2/Wen.Meng as:
```
[Wen.Meng@hfe06 ~/stmp2]$ ls -ltr */*diff
-rw-r--r-- 1 Wen.Meng stmp 66000 Oct 10 17:27 rtma_2023040400_pe_test/PRSLEV00.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp    75 Oct 10 17:28 rap_2020072316_pe_test/WRFPRS.GrbF16.diff
-rw-r--r-- 1 Wen.Meng stmp 65046 Oct 10 17:29 fv3r_2023062800_pe_test/PRSLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 65046 Oct 10 17:30 fv3r_2023062800/PRSLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 66000 Oct 10 17:30 rtma_2023040400/PRSLEV00.tm00.diff
```

**Comment by EricJames-NOAA** [comment]
@WenMeng-NOAA thanks for your tests, the results look good.  It was brought up at the RRFS meeting this morning that we need to make sure that the calculation works for the NSSL microphysics scheme.  So I may need to do a bit more testing.  

Or, alternatively, we can merge this PR and I can open a further one to get it working for the NSSL microphysics.  I will not have time to work on it until next week as I am defending my PhD on Friday.

**Comment by WenMeng-NOAA** [comment]
> @WenMeng-NOAA thanks for your tests, the results look good. It was brought up at the RRFS meeting this morning that we need to make sure that the calculation works for the NSSL microphysics scheme. So I may need to do a bit more testing.
> 
> Or, alternatively, we can merge this PR and I can open a further one to get it working for the NSSL microphysics. I will not have time to work on it until next week as I am defending my PhD on Friday.

@EricJames-NOAA Wen could hold on this PR until your testing is completed. No rush for your work.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA You may start the UPP RTs on Hera and Orion. Please refer to my testing on Hera /scratch1/NCEPDEV/stmp2/Wen.Meng as:
> 
> ```
> [Wen.Meng@hfe06 ~/stmp2]$ ls -ltr */*diff
> -rw-r--r-- 1 Wen.Meng stmp 66000 Oct 10 17:27 rtma_2023040400_pe_test/PRSLEV00.tm00.diff
> -rw-r--r-- 1 Wen.Meng stmp    75 Oct 10 17:28 rap_2020072316_pe_test/WRFPRS.GrbF16.diff
> -rw-r--r-- 1 Wen.Meng stmp 65046 Oct 10 17:29 fv3r_2023062800_pe_test/PRSLEV10.tm00.diff
> -rw-r--r-- 1 Wen.Meng stmp 65046 Oct 10 17:30 fv3r_2023062800/PRSLEV10.tm00.diff
> -rw-r--r-- 1 Wen.Meng stmp 66000 Oct 10 17:30 rtma_2023040400/PRSLEV00.tm00.diff
> ```

I understand we're holding on testing, but I do want to confirm with the current RT run that my results indeed match yours on Hera and Orion.

**Comment by EricJames-NOAA** [comment]
@WenMeng-NOAA I think I have the code working with the NSSL microphysics scheme now.  So I think this PR is ready for final review now.  Thanks for your patience.

**Comment by EricJames-NOAA** [comment]
Looping in @AndersJensen-NOAA

File: sorc/ncep_post.fd/MDLFLD.f [hunk 1]
```diff
@@ -96,8 +96,8 @@ SUBROUTINE MDLFLD
       use vrbls4d, only: dust, salt, suso, waso, soot, no3, nh4, smoke, fv3dust,&
               coarsepm
       use vrbls3d, only: zmid, t, pmid, q, cwm, f_ice, f_rain, f_rimef, qqw, qqi,&
-              qqr, qqs, cfr, cfr_raw, dbz, dbzr, dbzi, dbzc, qqw, nlice, nrain, qqg, zint, qqni,&
-              qqnr, qqnw, qqnwfa, qqnifa, uh, vh, mcvg, omga, wh, q2, ttnd, rswtt, &
+              qqr, qqs, cfr, cfr_raw, dbz, dbzr, dbzi, dbzc, qqw, nlice, nrain, qqg, qqh, zint,&
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/MDLFLD.f:100]: @EricJames-NOAA Please add a change log in this subroutine.

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Do you have sample model output files including new variable "hail" for my testing?

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Do you have sample model output files including new variable "hail" for my testing?

@WenMeng-NOAA I have sample files here on Hera:
/scratch1/BMC/wrfruc/ejames/dynf003-00-00.nc
/scratch1/BMC/wrfruc/ejames/phyf003-00-00.nc

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA See my test at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2022051200.
The new field is dumped as:
```
 wgrib2 PRSLEV03.tm00 -match VIL -s -stats
41:36907125:d=2022051200:VIL:entire atmosphere:3 hour fcst::ndata=1905141:undef=0:mean=0.108306:min=0:max=76.9344:cos_wt_mean=0.100289

```
Please let me know if you see issues in my test.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Can you rerun the UPP RTs on Hera and Orion since a new commit was added. You refer to my RT results at /scratch1/NCEPDEV/stmp2/Wen.Meng on Hera as:
```
[Wen.Meng@hfe10 ~/stmp2]$ ls -ltr */*diff
-rw-r--r-- 1 Wen.Meng stmp    75 Oct 17 23:45 rap_2020072316_pe_test/WRFPRS.GrbF16.diff
-rw-r--r-- 1 Wen.Meng stmp 66000 Oct 17 23:46 rtma_2023040400_pe_test/PRSLEV00.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 65046 Oct 17 23:46 fv3r_2023062800/PRSLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 65046 Oct 17 23:47 fv3r_2023062800_pe_test/PRSLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 66000 Oct 17 23:47 rtma_2023040400/PRSLEV00.tm00.diff
```

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Would you mind combining a minor change in your PR to remove a legacy UPP control file "postxconfig-NT-gefs-aerosol.txt" under UPP/parm? That is for addressing the UPP issue #804. Thanks!

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA See my test at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2022051200. The new field is dumped as:
> 
> ```
>  wgrib2 PRSLEV03.tm00 -match VIL -s -stats
> 41:36907125:d=2022051200:VIL:entire atmosphere:3 hour fcst::ndata=1905141:undef=0:mean=0.108306:min=0:max=76.9344:cos_wt_mean=0.100289
> ```
> 
> Please let me know if you see issues in my test.

@WenMeng-NOAA your test results are consistent with mine.  Thanks!

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Would you mind combining a minor change in your PR to remove a legacy UPP control file "postxconfig-NT-gefs-aerosol.txt" under UPP/parm? That is for addressing the UPP issue #804. Thanks!

No problem!  I just committed that change.

**Comment by WenMeng-NOAA** [comment]
> > @EricJames-NOAA Would you mind combining a minor change in your PR to remove a legacy UPP control file "postxconfig-NT-gefs-aerosol.txt" under UPP/parm? That is for addressing the UPP issue #804. Thanks!
> 
> No problem! I just committed that change.

@EricJames-NOAA Thank you!

**Comment by WenMeng-NOAA** [comment]
The UPP RTs were completed on WCOSS2. The new baselines for fv3r and 3drtma will be created.

**Comment by FernandoAndrade-NOAA** [comment]
@WenMeng-NOAA My Hera test results are consistent with yours on 3drtma and fv3r, Orion disk space is unfortunately still exceeded, so I can't run or even clone anything there at the moment.

**Review by FernandoAndrade-NOAA — APPROVED** [review]
Approved as Hera UPP RT results are consistent with Wen's Hera UPP RTs, Orion is currently unavailable for testing on my side due to space quota exceeded. Merge decision up for discussion.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Since the testing on WCOSS2 and Hera were completed, we may process the merging. You can recreated baselines on Orion when the disk space is available.

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA One more change might be missing. Since a new variable QQH is added, deallocation of it is needed in DEALLOCATE.f, such as:
deallocate(QQG)

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA One more change might be missing. Since a new variable QQH is added, deallocation of it is needed in DEALLOCATE.f, such as: deallocate(QQG)

@WenMeng-NOAA thanks for catching that!  I just made that change.

**Review by WenMeng-NOAA — APPROVED** [review]

**Pull request merged**